app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt

```diff
@@ -24,12 +28,86 @@ class TopicSummaryViewModel(
   val backgroundColor: Int = retrieveBackgroundColor()
   @ColorInt
   val darkerBackgroundOverlayColor: Int = computeDarkerBackgroundColor()
+  private val marginTopBottom = (activity as Context).resources
+      .getDimensionPixelSize(R.dimen.topic_list_item_margin_top_bottom)
+  private val marginMax = (activity as Context).resources.getDimensionPixelSize(R.dimen.home_margin_max)
+  private val marginMin = (activity as Context).resources.getDimensionPixelSize(R.dimen.home_margin_min)
+  private var position = -1
+  private var spanCount = activity.resources.getInteger(R.integer.home_span_count)
 
   /** Callback from data-binding for when the summary tile is clicked. */
   fun clickOnSummaryTile() {
     topicSummaryClickListener.onTopicSummaryClicked(topicSummary)
   }
 
+  fun setPosition(newPosition: Int) {
+    this.position = newPosition
+  }
+
+  fun computeTopMargin(): Int {
+    return marginTopBottom
+  }
+
+   fun computeBottomMargin(): Int {
+    return marginTopBottom
+  }
+
+  fun computeStartMargin(): Int {
+    var margin  = 0
+    when (spanCount) {
+      2 -> {
+        when (position % spanCount) {
+          0 -> margin = marginMin
+          else -> margin = marginMax
+        }
+      }
+      3 -> {
+        when (position % spanCount) {
+          0 -> margin = marginMax
+          1 -> margin = marginMin
+          2 -> margin = 0
+        }
+      }
+      4 -> {
+        when ((position + 1) % spanCount) {
+          0 -> margin = marginMax
+          1 -> margin = marginMin
+          2 -> margin = marginMin / 2
+          3 -> margin = 0
+        }
+      }
+    }
+    return margin  // error here?
+  }
+
+  fun computeEndMargin(): Int {
+    var margin = 0
+    when (spanCount) {
+      2 -> {
+        when (position % spanCount) {
+          0 -> margin = marginMax
+          else -> margin = marginMin
+        }
+      }
+      3 -> {
+        when (position % spanCount) {
+          0 -> margin = 0
+          1 -> margin = marginMin
+          2 -> margin = marginMax
+        }
+      }
+      4 -> {
+        when ((position + 1) % spanCount) {
+          0 -> margin = 0
+          1 -> margin = marginMin / 2
+          2 -> margin = marginMin
+          3 -> margin = marginMax
+        }
+      }
+    }
+    return margin // error here?
```

Comment:
```suggestion
    return when (spanCount) {
      2 -> when (position % spanCount) {
        0 -> marginMax
        else -> marginMin
      }
      3 -> when (position % spanCount) {
        0 -> 0
        1 -> marginMin
        2 -> marginMax
        // is else needed here? If so, maybe default to 0.
      }
      4 -> when ((position + 1) % spanCount) {
        0 -> 0
        1 -> marginMin / 2
        2 -> marginMin
        3 -> marginMax
        // ditto here for else (Kotlin compiler might not be smart enough to realize this is an exhaustive when)
      }
      else -> 0
    }
```

This can be simplified & made to avoid using a variable.

Reply:
Done